Library/Homebrew/os/mac/universal_server.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require "extend/ENV"
+require "fcntl"
+require "socket"
+require "os/universal_server"
+
+class UniversalServer
+  def self.open
+    Dir.mktmpdir("homebrew", HOMEBREW_TEMP) do |tmpdir|
```

Comment:
Given this is identical on macOS and Linux it should just live in `Library/Homebrew/universal_server.rb` (with a better name like `Server`)